src/main.rs

```diff
@@ -206,6 +280,241 @@ impl BFInterp {
         }
         string
     }
+
+    fn int_bin_operations(
+        &mut self,
+        op0: &Operand,
+        op1: &Operand,
+        dest: &name::Name,
+        operation_type: BinOps,
+    ) {
+        match op0 {
+            LocalOperand { name, ty } => match ty.as_ref() {
+                Type::IntegerType { bits } => match bits {
+                    8 => todo!(),
+                    16 => todo!(),
+                    32 => {
+                        let op0 = self.get_int_op(self.vars.get(name).unwrap()) as i32;
+                        let op1 = self.get_int_op(op1) as i32;
+                        self.int_bin_operation(op0, op1, dest, operation_type)
+                    }
+                    64 => {
+                        let op0 = self.get_int_op(self.vars.get(name).unwrap()) as i64;
+                        let op1 = self.get_int_op(op1) as i64;
+                        self.int_bin_operation(op0, op1, dest, operation_type)
+                    }
+                    128 => todo!(),
+                    _ => todo!(),
+                },
+                _ => todo!(),
+            },
+            ConstantOperand(con_op) => match con_op.as_ref() {
+                Constant::Int { .. } => todo!(),
+                Constant::Vector(_) => todo!(),
+                _ => todo!(),
+            },
+            MetadataOperand => todo!(),
+        }
+    }
+
+    fn get_int_op(&self, op: &Operand) -> u64 {
+        match op {
+            LocalOperand { name, .. } => self.get_int_op(self.vars.get(name).unwrap()),
+            ConstantOperand(con_op) => match con_op.as_ref() {
+                Constant::Int { value, .. } => *value,
+                Constant::Vector(_) => todo!(),
```

Comment:
All of the other variants should be `unreachable!()` I think, because this function assumes that the operand is an integer.

Reply:
I haven't considered the case where we have a vector of integers which would also use the int_bin_operations() method but for now I'll change all to unreachable (and take out the vector arm) except for Int obviously and Undef. I'm not sure what to leave Undef as. It's currently a panic!(). It will go to the Undef arm if for example a divisor is 0.